scripts/make-guac.sh

```diff
@@ -422,6 +493,44 @@ then
     docker rm "${DOCKER_GUACAMOLE}" | log
 fi
 
+# Initialize arrays for Docker run parameters
+params_begin=(
+    --restart unless-stopped
+    --link guacd:guacd
+    -v "${GUAC_HOME}":/guac-home
+    -e GUACAMOLE_HOME=/guac-home
+)
+
+params_saml=()
+
+params_mysql=()
+
+params_end=(
+    -d -p 8080:8080 "${DOCKER_GUACAMOLE_IMAGE}"
+)
+
+#[[ -n $LDAP_HOSTNAME ]] && params+=(-e LDAP_HOSTNAME=${LDAP_HOSTNAME})
+#[[ -n $LDAP_USER_BASE ]] && [[ -n $LDAP_DOMAIN_DN ]] && params+=(-e LDAP_USER_BASE_DN="${LDAP_USER_BASE},${LDAP_DOMAIN_DN}")
+#[[ -n $LDAP_USER_ATTRIBUTE ]] && params+=(-e LDAP_USER_ATTRIBUTE=${LDAP_USER_ATTRIBUTE})
+#[[ -n $LDAP_CONFIG_BASE ]] && [[ -n $LDAP_DOMAIN_DN ]] && params+=(-e LDAP_CONFIG_BASE_DN="${LDAP_CONFIG_BASE},${LDAP_DOMAIN_DN}")
+#[[ -n $LDAP_GROUP_BASE ]] && [[ -n $LDAP_DOMAIN_DN ]] && params+=(-e LDAP_GROUP_BASE_DN="${LDAP_GROUP_BASE},${LDAP_DOMAIN_DN}")
+#[[ -n $LDAP_PORT ]] && params+=(-e LDAP_PORT=${LDAP_PORT})
+
+# Build SAML parameters if present
+[[ -n $SAML_IDP_METADATA_URL ]] && params_saml+=(-e SAML_IDP_METADATA_URL=${SAML_IDP_METADATA_URL})
```

Comment:
I'm pretty sure this construct fails when using `set -e` because the "or" condition is an errexit. Might need to flip the condition....

```suggestion
[[ -z $SAML_IDP_METADATA_URL ]] || params_saml+=(-e SAML_IDP_METADATA_URL=${SAML_IDP_METADATA_URL})
```

Reply:
Can you clarify?  Your suggestion is to use the "or" condition.

Reply:
Using `&&` handles the success condition, performing whatever the action is. But when the test `[[ ... ]]` fails, it errexits. The success condition is skipped and the command will end with the errexit. When `set -e` is in place, that will terminate the script. A solution I've used is to flip the logic, and make the errexit the condition when the action is performed, and a success exit is just a noop. Hence,

```
[[ <test> ]] || <action>
```

That setup can be read as, "If the test _fails_, perform the action. If the action succeeds, exit success. If the action fails, errexit. When the test succeeds, do nothing."